### DISCUSSION TITLE
API Keys in android build?

### DISCUSSION THREAD

**Discussion by JohnessyAI** [srcnalt/OpenAI-Unity#55] [Q&A]
Hi there, 

Is there anything in particular i need to do to include the keys here? I did think about using Firebase to grab them and then set them but dont want to go do that if there is any easier way ?

**Comment by srcnalt** [comment]
Hi @JohnessyAI, keeping it hardcoded and retrieving from Firebase is almost the same, at the end API Key will be used in client side. The best approach is delegating the whole operation to a backend without exposing the API Key and just sending and receiving json data.

**Reply by JohnessyAI** [reply]
Thanks for this! 

Can you explain how you would do it? I have tried hard coding it in but for some reason it still goes to the .openai folder. Would love your brains on this. BTW Your system is amazing and i want to say thank you so much!

**Reply by srcnalt** [reply]
You welcome @JohnessyAI glad you found it useful. At the moment you can pass the API key into OpenAiAPI instance l like the following. Then it would ignore .openai folder and just use the entered parameter string.

```c#
var openai = new OpenAIApi("sk-Me8...6yi");
```
